pkg/apis/pipeline/v1beta1/pipelinerun_validation.go

```diff
@@ -101,3 +133,21 @@ func (ps *PipelineRunSpec) Validate(ctx context.Context) (errs *apis.FieldError)
 
 	return errs
 }
+
+func (ps *PipelineRunSpec) validatePipelineTimeout(timeout time.Duration, errorMsg string) (errs *apis.FieldError) {
+	if ps.Timeouts.Tasks != nil && ps.Timeouts.Tasks.Duration > timeout {
+		errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%s "+errorMsg, ps.Timeouts.Tasks.Duration.String()), "timeouts.tasks"))
+	}
+
+	if ps.Timeouts.Finally != nil && ps.Timeouts.Finally.Duration > timeout {
+		errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%s "+errorMsg, ps.Timeouts.Finally.Duration.String()), "timeouts.finally"))
```

Comment:
small nit: mixing `+` string concatenation with `fmt.Sprintf`.  suggest sticking with one or other. e.g.

```go
fmt.Sprintf("%s %s", ps.Timeouts.Finally.Duration.String(), errorMsg)
```

I tend to prefer this since it tells me immediately that `errorMsg` isn't allowed to contain formatting sequences of its own. Purely a code style thing though if you want to stick with the existing.